docs/user-guide/app_dependencies.md

```diff
@@ -0,0 +1,133 @@
+# Application dependencies
+
+Argo CD lets you natively model dependencies between Applications. This mechanism is intended to replace the prior way of modeling dependencies when using the app-of-apps pattern with sync waves on the Application custom resources.
```

Comment:
I think we probably need a full section describing the pros/cons of different ordering mechanisms: appset progressive syncs, app-of-apps w/ sync waves, app dependencies, and external tooling like CI / Kargo / Telefonistka. I'm not yet sure that app-of-apps w/ sync waves should be considered an anti-pattern. If it is, we should have a docs section explaining why.